ang/civicase/ActivityCard.js

```diff
@@ -185,14 +294,32 @@
         return crmBlocker(crmStatus({start: $scope.ts('Deleting...'), success: $scope.ts('Deleted')}, p));
       };
     };
-  }
 
-  /**
-   * Activity Card link function
-   *
-   * @param {Object} $scope
-   */
-  function caseActivityCardLink ($scope) {
-    $scope.bootstrapThemeElement = $('#bootstrap-theme');
+    /**
+     * Set Allowed Activity status's
+     */
+    function setAllowedActivityStatuses () {
+      $scope.allowedActivityStatuses = {};
+
+      _.each(CRM.civicase.activityStatuses, function (activityStatus, activityStatusID) {
+        var ifStatusIsInSameCategory = _.intersection($scope.activity.category, activityStatus.grouping.split(',')).length > 0;
+        var ifStatusIsInNoneCategory = $scope.activity.category.length === 0 && activityStatus.grouping.split(',').indexOf('none') !== -1;
+
+        if (ifStatusIsInSameCategory || ifStatusIsInNoneCategory) {
+          $scope.allowedActivityStatuses[activityStatusID] = activityStatus;
+        }
+      });
+    }
```

Comment:
Can be refactored as

```js
$scope.allowedActivityStatuses = _.pickBy(CRM.civicase.activityStatuses, function (activityStatus) {
  var ifStatusIsInSameCategory = _.intersection($scope.activity.category, activityStatus.grouping.split(',')).length > 0;
  var ifStatusIsInNoneCategory = $scope.activity.category.length === 0 && activityStatus.grouping.split(',').indexOf('none') !== -1;

  return ifStatusIsInSameCategory || ifStatusIsInNoneCategory
});
```

`pickBy` is the same as `filter`, but it returns a filtered object instead of an array.

Reply:
`pickBy`does not work in our version of lodash :(